### PR TITLE
Status displays have their text color update when the alert level changes.

### LINKED ISSUE #8642: Status Display text should change color based on the alert level.
### Description

Currently, it can be very difficult to tell what alert level the station is at if you miss the alert given in chat. An effective and easy way to better communicate the situation would be to change the always blue text on the status displays to mimic the alert level, so, green, blue, or red text. code delta, being black, wouldn't be very legible on the dark displays, so perhaps purple or blinking red?

perhaps some examination text on the Status Display would help as well. It could be useful if the Status Display could give a lot of circumstantial information on examination, actually, as they are a go to source for _status_ related stuff.

### PR DESCRIPTION
**Pull request by MaxIsJoe**
closes #8642

CL: [New] - Status displays have their text color update when the alert level changes.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Objects/StatusDisplay.cs [hunk 1]
```diff
@@ -99,6 +102,7 @@ public void OnSpawnServer(SpawnInfo info)
 			SyncSprite(stateSync, stateSync);
 			centComm = GameManager.Instance.CentComm;
 			centComm.OnStatusDisplayUpdate.AddListener(OnTextBroadcastReceived);
+			centComm.OnAlertLevelChange += UpdateTextColor;
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Objects/StatusDisplay.cs]: this needs to be networked

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Objects/StatusDisplay.cs]: probably a syncvar with current colour would be best

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Objects/StatusDisplay.cs]: likie dis?

File: UnityProject/Assets/Scripts/Objects/StatusDisplay.cs [hunk 2]
```diff
@@ -290,8 +299,14 @@ private IEnumerator BlinkText()
 			this.StartCoroutine(BlinkText(), ref blinkHandle);
 		}
 
+		private void UpdateTextColor()
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Objects/StatusDisplay.cs]: Need the two colour  thingy old and new

File: UnityProject/Assets/Scripts/Objects/StatusDisplay.cs [hunk 3]
```diff
@@ -290,8 +299,14 @@ private IEnumerator BlinkText()
 			this.StartCoroutine(BlinkText(), ref blinkHandle);
 		}
 
+		private void UpdateTextColor(Color oldValue, Color newValue)
+		{
+			textField.color = centComm.CurrentAlertLevel == CentComm.AlertLevel.Red ? redAlertTextColor : normalTextColor;
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Objects/StatusDisplay.cs]: This is in the client though how do they know what CurrentAlertLevel is?

 That's why you use the Color oldValue, Color newValue) on client

**Comment by corp-0** [comment]
seems like it isn't compiling

**Comment by MaxIsJoe** [comment]
opening rider in 5

**Comment by MaxIsJoe** [comment]
done

**Review by NoooneyDude — APPROVED** [review]

File: UnityProject/Assets/Scripts/Objects/StatusDisplay.cs [hunk 4]
```diff
@@ -290,8 +299,19 @@ private IEnumerator BlinkText()
 			this.StartCoroutine(BlinkText(), ref blinkHandle);
 		}
 
+		private void ServerUpdateCurrentColor()
+		{
+			currentTextColor = centComm.CurrentAlertLevel == CentComm.AlertLevel.Red ? redAlertTextColor : normalTextColor;
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Objects/StatusDisplay.cs]:
```suggestion
			currentTextColor = centComm.CurrentAlertLevel == CentComm.AlertLevel.Red || centComm.CurrentAlertLevel == CentComm.AlertLevel.Delta
				? redAlertTextColor
				: normalTextColor;
```

**Review by NoooneyDude — CHANGES_REQUESTED** [review]

**Comment by MaxIsJoe** [comment]
Done 👍

**Review by ThatDan123 — APPROVED** [review]

**Pull request merged**